src/org/labkey/targetedms/outliers/OutlierGenerator.java

```diff
@@ -344,6 +371,24 @@ private void retainIfPreferred(Map<String, QCMetricConfiguration> preferredConfi
         return precursors;
     }
 
+    private boolean isExcludedPrecursorPeptide(Collection<Map<String, Object>> excludedPrecursors, String modifiedSeq, int charge, double mz)
+    {
+        String peptidePrecursorIdentifier = modifiedSeq + "," + charge + "," + DECIMAL_FORMAT.format(mz);
```

Comment:
This seems fragile since it needs to match up with how the SQL query is building the string. Can we find a way to consolidate implementations and/or add comments saying that they have to be kept in sync.

Reply:
Now re-implemented.